libdnf/dnf-context.cpp

```diff
@@ -3481,12 +3483,14 @@ dnf_context_module_install(DnfContext * context, const char ** module_specs, GEr
                         } else {
                             profiles.push_back(modpkg->getDefaultProfile());
                         }
+                        auto nsvca = modpkg->getFullIdentifier();
```

Comment:
I suggest that there is an issue in workflow. Before we can look at profiles modular sack must be resolved. If I am not mistaken call `auto solver_error = recompute_modular_filtering(container, sack, hotfixRepos);` is to late. Then I should take only the latest active module to collect profiles for install.

Reply:
There is also a problem that modules or NSVCA is not installable. Only profiles that contains names of packages. can be installed.

Reply:
Thanks for the feedback. So in other words roughly what you're suggesting is this order:
- `modify_module_dict_and_enable_stream()`
- `recompute_modular_filtering()`
- `container->install()` profiles we want
- `dnf_context_install()` for artifacts of each profile

?

But doesn't `modify_module_dict_and_enable_stream()` already activate the right modules from which we should be searching for profiles? Testing this code with https://github.com/coreos/rpm-ostree/pull/2760, it seems to work well. Though I haven't tested all the corner cases.

> There is also a problem that modules or NSVCA is not installable. Only profiles that contains names of packages. can be installed.

Yes, we're only installing at the profile level here. This is following the same semantics as `dnf module install`, where it auto-fills the stream (if there is only one) and the profile (if there is a default one).

Reply:
I guess that there will be a difference when modules in the same stream will have a different set of profiles or profiles will have a different set of rpms names. I can provide a real examples if you require it.

I suggest that we have to resolve the workflow issue first and then we can look for a solution that was original target of PR. Do you agree?

Reply:
Hi @j-mracek, thanks for the feedback. I reworked things now to make this more or less in line with what `dnf module install` does. Consequently, we now correctly depsolve the modular sack before looking at profiles and packages.